src/manager/lotus.rs

```diff
@@ -188,23 +188,57 @@ impl<T: JsonRpcClient + Send + Sync> SubnetManager for LotusSubnetManager<T> {
         Ok(())
     }
 
-    async fn propagate(
-        &self,
-        _subnet: SubnetID,
-        _from: Address,
-        _postbox_msg_cid: Cid,
-    ) -> Result<()> {
-        todo!()
+    async fn propagate(&self, subnet: SubnetID, from: Address, postbox_msg_cid: Cid) -> Result<()> {
+        if !self.is_network_match(&subnet).await? {
+            return Err(anyhow!("propagation not targeting the correct network"));
+        }
+
+        let params = cbor::serialize(
+            &PropagateParams {
+                postbox_cid: postbox_msg_cid,
+            },
+            "propagate params",
+        )?;
+
+        let message = MpoolPushMessage::new(
+            Address::new_id(DEFAULT_IPC_GATEWAY_ADDR),
+            from,
+            ipc_gateway::Method::Propagate as MethodNum,
+            params.to_vec(),
+        );
+
+        self.mpool_push_and_wait(message).await?;
```

Comment:
@cryptoAtwill , something I was wondering here while I was doing tests. Is there any timeout for this await? Or if the message stays pending forever this request stays pending forever? I guess the rpc server has some kind of timeout but I wanted to double-check.

Reply:
Currently this wait has no timeout, at least in the agent. We can introduce a timeout, but I guess the question is really what do we do when it times out, we just return an error? But I think it would be a better practice to have a timeout.